src/lib/comms/sol-socket-dtls.h

```diff
@@ -37,7 +37,7 @@ enum sol_socket_dtls_cipher {
 struct sol_socket_dtls_credential_cb {
     const void *data;
 
-    void *(*init)(const void *data);
+    int (*init)(const void *data);
```

Comment:
I don't get it. We don't need a size and the data here is a pointer to user data. It is initializing a psk/key store. init and clear are not being used anymore, but I kept them because they make sense.